test/run_envoy_tests.sh

```diff
@@ -61,10 +61,16 @@ then
   exit 0
 fi
 
+# TODO(htuch): Clean this up when Bazelifying the hot restart test below.
+HOT_RESTART_JSON="$TEST_SRCDIR"/test/config/integration/hot_restart.json
+cat "$TEST_TMPDIR"/test/config/integration/server.json |
```

Comment:
Yeah I kind of like the idea of just spitting the JSON back out, or just generally having some way to have an admin endpoint that maybe spits out the listener list in JSON with the real ports. I think this is complex enough to fix and the risk of regression is low enough that I'm fine deferring this for now. We should open a GH issue though that specifically tracks various tech debt cleanups (or just track specifically in the making build better issue) from the migration just to make sure that we don't lose them in "TODO noise".